lib/surgex/types.ex

```diff
@@ -0,0 +1,5 @@
+defmodule Surgex.Types do
+  @moduledoc false
+
+  @type json_value :: number | String.t() | nil | list | map
+end
```

Comment:
Awesome 👏🏻

Reply:
I think `Surgex.Parser` isn't used only for JSON values, it's also used for URL params. In general, I'd say you can pass any term to the parser but only select terms will be parsed correctly, all others will return an error. So maybe just use `term()`?

Reply:
I guess it depends on how we treat typespecs. Sure, parser won't technically break on other values, but having any type other than these (URL param values are a subset of JSON values if I'm not mistaken) means that the parser is used incorrectly. Type check could find that and notify the programmer. If we use `term()`, this information would be lost.

However if we treat typespecs as information that nothing shall break if the spec is met then yes, using `term()` is fine.